### PR TITLE
[PHPStanStaticTypeMapper] Handle crash on DowngradeMixedTypeDeclarationRector for PHPStan\Type\ConditionalType

### LINKED ISSUE #8099: System error: "Rector\PHPStanStaticTypeMapper\PHPStanStaticTypeMapper::mapToPHPStanPhpDocTypeNode
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | latest dev-main             |

Rector is producing this error:

```json
{
    "totals": {
        "changed_files": 0,
        "errors": 1
    },
    "errors": [
        {
            "message": "System error: \"Rector\\PHPStanStaticTypeMapper\\PHPStanStaticTypeMapper::mapToPHPStanPhpDocTypeNode for PHPStan\\Type\\ConditionalType\"\nRun Rector with \"--debug\" option and post the report here: https://github.com/rectorphp/rector/issues/new",
            "file": "/var/www/src/../storage/demo/nhiokqbj8b3in8kegccs/rector_analyzed_file.php",
            "line": 36
        }
    ]
}
```

## Minimal PHP Code Causing Issue

The error happens with the following phpdoc (from [class `ParameterBag` in Symfony DependencyInjection](https://github.com/symfony/symfony/blob/7c0b00c86148428c3d732802db3fb69380b057d5/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php#L171-L185)):

```php
<?php

namespace Symfony\Component\DependencyInjection\ParameterBag;

class ParameterBag
{
    /**
     * @template TValue of array<array|scalar>|scalar
     *
     * @param TValue $value
     *
     * @return (TValue is scalar ? array|scalar : array<array|scalar>)
     */
    public function resolveValue(mixed $value, array $resolving = []): mixed
    {
        return '';
    }
}
```

The problem happens with `DowngradeLevelSetList::DOWN_TO_PHP_74`:

https://getrector.com/demo/ccaae362-0585-4dfe-8136-02bd85684fc2

And it does not happen with `DowngradeLevelSetList::DOWN_TO_PHP_80`:

https://getrector.com/demo/a5718e52-8605-42c9-ac2a-2cc29fb9c870

I also checked in PHPStan, and it seems to work well there:

https://phpstan.org/r/718555a5-2550-4001-86e8-e55f5ee43670

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8099

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba I added `ConditionalTypeMapper`, it currently produce invalid result:

```diff
     /**
      * @template TValue of array<array|scalar>|scalar
      *
-     * @return (TValue is scalar ? array|scalar : array<array|scalar>)
+     * @return (Rector\Core\Tests\Issues\IssueConditionalType\Fixture\TValue is (bool | float | int | string) ? (array | bool | float | int | string) : array<(array | bool | float | int | string)>)
      */
```

File: packages/PHPStanStaticTypeMapper/TypeMapper/ConditionalTypeMapper.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
+    {
+        $type = TypeTraverser::map($type, static function (Type $type, callable $traverse): Type {
+            if ($type instanceof \PHPStan\Type\ObjectType && $type->getClassReflection() === null) {
+                $newClassName = Strings::after($type->getClassName(), '\\', -1);
+                $type = new \PHPStan\Type\ObjectType($newClassName);
+            }
+
+            return $traverse($type);
+        });
```
Comment (samsonasik) [inline packages/PHPStanStaticTypeMapper/TypeMapper/ConditionalTypeMapper.php]: @TomasVotruba this is to tweak the `@template` usage

**Comment by samsonasik** [comment]
@leoloso it fixed, with tweak, I am not sure if this will be an elegant solution, but just looking at the use case, it seems the way to resolve it.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thanks 🙏

**Comment by leoloso** [comment]
@samsonasik it's working well, thanks! 🙏

**Pull request merged**